Barotrauma/BarotraumaShared/Source/Characters/StatusEffect.cs

```diff
@@ -138,7 +141,20 @@ protected StatusEffect(XElement element)
                             " - sounds should be defined as child elements of the StatusEffect, not as attributes.");
                         break;
                     default:
-                        propertyAttributes.Add(attribute);
+                        object attributeObject = XMLExtensions.GetAttributeObject(attribute);
+                        Type type = attributeObject.GetType();
+
+                        string op = ((string)attributeObject).Substring(0, 2);
```

Comment:
This will throw an exception if the value of the attribute is a number, a boolean or a too short string. Also, you should probably use GetAttributeString because you're expecting the value to be a string.

Reply:
But I can't use GetAttributeString because "attribute" not an XElement, it's an XAttribute. Maybe I should do something like `string op = (attributeObject.ToString()).Substring(0, 2);` instead?

Reply:
Ah yeah you're right, GetAttributeString won't work here. There's still no reason to parse the value of the attribute to an object and then cast it back to a string - if you want the string representation of the value you could simly use attribute.Value. And make sure it's long enough before calling substring.